includes/wikia/services/UserService.class.php

```diff
@@ -21,6 +18,29 @@ public static function getNameFromUrl($url) {
 	}
 
 	/**
+	 * get main page for current user respecting user preferences
+	 * @return Title
+	 */
+	public static function getMainPage() {
+		global $wgUser;
+
+		$title = Title::newMainPage();
+
+		if( $wgUser->isLoggedIn() ) {
+			$value = $wgUser->getGlobalPreference(UserPreferencesV2::LANDING_PAGE_PROP_NAME);
+			switch($value) {
+				case UserPreferencesV2::LANDING_PAGE_WIKI_ACTIVITY:
+					$title = SpecialPage::getTitleFor('WikiActivity');
+					break;
+				case UserPreferencesV2::LANDING_PAGE_RECENT_CHANGES:
+					$title = SpecialPage::getTitleFor('RecentChanges');
+					break;
+			}
+		}
```

Comment:
wouldn't be nicer to add here the default, as sonar suggests with `$title = Title::newMainPage();` instead of doing it in l27?